internal/blocktx/processor.go

```diff
@@ -188,45 +187,56 @@ func (p *Processor) StartBlockProcessing() {
 			case <-p.ctx.Done():
 				return
 			case blockMsg := <-p.blockProcessCh:
+				blockhash := blockMsg.Header.BlockHash()
+				p.logger.Info("received block", slog.String("hash", blockhash.String()))
+
 				err := p.processBlock(blockMsg)
+
 				if err != nil {
-					blockhash := blockMsg.Header.BlockHash()
-					_, errDel := p.store.DelBlockProcessing(p.ctx, &blockhash, p.hostname)
-					if errDel != nil {
-						p.logger.Error("failed to delete block processing", slog.String("hash", blockhash.String()), slog.String("err", errDel.Error()))
-					}
+					p.logger.Error("block processing failed", slog.String("hash", blockhash.String()), slog.String("err", err.Error()))
+					p.unlockBlock(p.ctx, &blockhash)
 				}
 			}
 		}
 	}()
 }
 
-func (p *Processor) StartFillGaps(peers []p2p.PeerI) {
+// unlock block for future processing
+func (p *Processor) unlockBlock(ctx context.Context, hash *chainhash.Hash) {
+	// use closures for retries
+	unlockFn := func() error {
+		_, err := p.store.DelBlockProcessing(ctx, hash, p.hostname)
+		return err
+	}
+
+	if unlockErr := withRetry(ctx, unlockFn, 5); unlockErr != nil {
```

Comment:
Let's put that `5` to the consts on top, maybe `unlockBlockRetries` or something